### PR TITLE
DAOS-6304 integration: harmonize user/group setup

### PR DESCRIPTION
**Pull request by Michael-Hennecke**
In gen_certificates.sh, add explicit "chown user.group" to cert files

Signed-off-by: Michael Hennecke <mhennecke@lenovo.com>

### CODE REVIEW

**Review by mjmac — CHANGES_REQUESTED** [review]
This script is intended to be used by developers as well, and shouldn't require root access. I think a slight rework is needed to only do the ownership changes if EUID is 0.

**Comment by Michael-Hennecke** [comment]
> 
> 
> This script is intended to be used by developers as well, and shouldn't require root access. I think a slight rework is needed to only do the ownership changes if EUID is 0.

I thought the 2>/dev/null was taking care of the case of not running as root (as well as the situation where the user or group may be non-existent). Pushed another commit that checks for EUID 0 before running chmod to make this cleaner.

**Review by johannlombardi — APPROVED** [review]

File: utils/certs/gen_certificates.sh [hunk 1]
```diff
@@ -206,6 +221,8 @@ function generate_test_cert () {
         -cert "${CERTS}/daosCA.crt" -policy signing_policy \
         -extensions signing_test -out "${CERTS}/test.crt" \
         -outdir "${CERTS}" -in "${CA_HOME}/test.csr" -batch
+    # TODO [[ $EUID -eq 0 ]] && chown ?.? "${CERTS}/test.crt"
+    chmod 0644 "${CERTS}/test.crt"
```
Comment (johannlombardi) [inline utils/certs/gen_certificates.sh:225]: @brianjmurrell @dinghwah Could you please advise what user we should use for the test.crt?

**Comment by johannlombardi** [comment]
@dpquigl could you please review?

File: utils/certs/gen_certificates.sh (hunk 1 above)
Comment (brianjmurrell) [reply utils/certs/gen_certificates.sh:225]: I don't know what it's used for.  Perhaps @dinghwah could explain.

Comment (dinghwah) [reply utils/certs/gen_certificates.sh:225]: I do think we should remove test.crt and related lines now.
test.crt was used in the early phase for testing propose.

Comment (Michael-Hennecke) [reply utils/certs/gen_certificates.sh:225]: OK I will do that now and push again, thanks!

**Review by mjmac — APPROVED** [review]
LGTM. For the TODOs on admin certs, I'm not sure we can handle all cases in the script. Any user with access to the admin certificate is by definition an admin, but I think this is going to be very site-specific.

Would like to see a review by @dpquigl before we land this.

**Comment by Michael-Hennecke** [comment]
@dpquigl, any updates on this? It should not do any harm to land this as is now, and re-visit the admin role/user/group TODO comment at a later time...

**Review by dpquigl — COMMENTED** [review]
Umm this amkes it so you can't generate the certificates on a machine which doesn't have the agent and server installed..... Seems like a bad idea and makes it so you have to generate your certificates in a non-secure area.

**Comment by dpquigl** [comment]
As I was telling Mike I'd much rather see this done as a separate script. This change requires you to run the script on a machine with the daos server and client packages installed. Also you don't run this script on every machine you run it once and you'll then have to copy it over the network. Unless you're copying it as root (which is not advised and most systems don't even allow root ssh/scp access) then you're going to have to redo all of these permissions on the remote end anyway.

**Comment by Michael-Hennecke** [comment]
I don't think that the script requires you to run it on a host with daos server and client installed: If the daos users/groups don't exist, the chown will silently fail, and the change in this PR does exactly nothing. But if the users and groups exist, the script now sets them to the intended values. 
This script did already set the permissions of the files before this PR. By the same logic, those permissions also will need to be checked abd potentially re-set on the actual daos nodes when the certs are created on another system. But it helps that the script sets the permissions at creation time, because you can then explicitly see what the intended permissions are. 

I am following the same logic here for the user and group: What we want to achieve here is that it's explicit what the intended "user" and "group" permissions are. Without specifying a group, it's not really clear what a group permission means in the first case.

**Comment by dpquigl** [comment]
Sorry requires was perhaps a bit strong but as you pointed out without those packages installed this script change doesn't do anything. One of my concerns here is that since this script is setting permissions on the host it gives the impression that it should be run on a per-host basis which will cause a lot of confusion when the system doesn't work.  This is why I think an install script is a better idea since you can just package up the certs directory and the script and push it out to all of the nodes.

**Comment by dpquigl** [comment]
Also with scp you can preserve the mode but do not have a way to specify preserving the ownership information. That's why setting the mode on the files remotely is ok but not necessarily the ownership information.

**Comment by Michael-Hennecke** [comment]
Let's get on a call next week to resolve this...

**Comment by johannlombardi** [comment]
Let's merge master and land this PR. @dpquigl will create separate jira ticket for the certificate deployment.

**Comment by Michael-Hennecke** [comment]
i did a git pull/git push ... no changes to the files touched by this PR

**Review by dpquigl — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]
LGTM

**Comment by dpquigl** [comment]
For reference here is the link to the jira ticket for follow-on work.

https://jira.hpdd.intel.com/browse/DAOS-6694

**Pull request merged**